src/app/locator/qgsinbuiltlocatorfilters.h

```diff
@@ -151,6 +151,32 @@ class QgsBookmarkLocatorFilter : public QgsLocatorFilter
 
 };
 
+class QgsSettingsLocatorFilter : public QgsLocatorFilter
+{
+    Q_OBJECT
+
+  public:
+
+    QgsSettingsLocatorFilter( QObject *parent = nullptr );
+    QgsSettingsLocatorFilter *clone() const override;
+    QString name() const override { return QStringLiteral( "optionpages" ); }
+    QString displayName() const override { return tr( "Settings" ); }
+    Priority priority() const override { return Highest; }
+    QString prefix() const override { return QStringLiteral( "s" ); }
+    QgsLocatorFilter::Flags flags() const override { return QgsLocatorFilter::FlagFast; }
+
+    void fetchResults( const QString &string, const QgsLocatorContext &context, QgsFeedback *feedback ) override;
+    void triggerResult( const QgsLocatorResult &result ) override;
+
+    struct SettingsHandle
+    {
+      QString type;
+      QString page;
+    };
+};
+
+Q_DECLARE_METATYPE( QgsSettingsLocatorFilter::SettingsHandle )
```

Comment:
I think instead of metatype the struct here it's a better approach to use a QVariantMap with keys for type/page (which can be stored in a QVariant)